crates/miden-objects/src/account/builder/mod.rs

```diff
@@ -100,15 +120,22 @@ impl AccountBuilder {
         #[cfg(all(not(feature = "testing"), not(test)))]
         let vault = AssetVault::default();
 
-        let (code, storage) =
-            Account::initialize_from_components(self.account_type, &self.components).map_err(
-                |err| {
-                    AccountError::BuildError(
-                        "account components failed to build".into(),
-                        Some(Box::new(err)),
-                    )
-                },
-            )?;
+        let auth_component = self
+            .auth_component
+            .clone()
+            .ok_or(AccountError::BuildError("auth component must be set".into(), None))?;
```

Comment:
Nit: I would avoid the clone by changing the method to take `&mut self` and use `Option::take` instead.

Reply:
Done in [40b9b45](https://github.com/0xMiden/miden-base/pull/1480/commits/40b9b458b61965259ddde9afb060f76659c4c75a)